ofctrl/fgraphFlow.go

```diff
@@ -611,26 +605,23 @@ func getRangeEnd(rng *openflow13.NXRange) uint16 {
 }
 
 func merge(regs []*NXRegister) *NXRegister {
-	if len(regs) == 1 {
-		return regs[0]
-	}
-	var data uint32
-	min := regs[0].Range.GetOfs()
-	max := getRangeEnd(regs[0].Range)
+	var data, mask uint32
 	for _, reg := range regs {
-		data |= reg.Data << reg.Range.GetOfs()
-		end := getRangeEnd(reg.Range)
-		if reg.Range.GetOfs() < min {
-			min = reg.Range.GetOfs()
-		}
-		if end > max {
-			max = end
+		if reg.Range != nil {
+			end := getRangeEnd(reg.Range)
+			start := reg.Range.GetOfs()
+			data |= reg.Data << start
+			mask |= ((uint32(1) << (end - start + 1)) - 1) << start
+		} else {
+			// full range
+			data |= reg.Data
+			mask |= 0xffffffff
```

Comment:
What if reg.Mask is set but reg.Range==nil?

Reply:
Here we should use provided reg.Mask if it is set firstly. I have added this check in update.